vector/src/main/java/im/vector/app/features/onboarding/ftueauth/FtueAuthChooseDisplayNameFragment.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2021 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.onboarding.ftueauth
+
+import android.os.Bundle
+import android.text.Editable
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.google.android.material.textfield.TextInputLayout
+import im.vector.app.core.platform.SimpleTextWatcher
+import im.vector.app.databinding.FragmentFtueDisplayNameBinding
+import im.vector.app.features.onboarding.OnboardingAction
+import im.vector.app.features.onboarding.OnboardingViewEvents
+import javax.inject.Inject
+
+class FtueAuthChooseDisplayNameFragment @Inject constructor() : AbstractFtueAuthFragment<FragmentFtueDisplayNameBinding>() {
+
+    override fun getBinding(inflater: LayoutInflater, container: ViewGroup?): FragmentFtueDisplayNameBinding {
+        return FragmentFtueDisplayNameBinding.inflate(inflater, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setupViews()
+    }
+
+    private fun setupViews() {
+        views.displayNameSubmit.isEnabled = views.displayNameInput.hasContentEmpty()
+        views.displayNameInput.editText?.addTextChangedListener(object : SimpleTextWatcher() {
+            override fun afterTextChanged(s: Editable) {
+                val newContent = s.toString()
+                views.displayNameSubmit.isEnabled = newContent.isNotEmpty()
```

Comment:
Could be nice to reset the error here.
Because the error when setting display name could be displayed in the TextInputLayout. You can trigger an error for instance by setting a display name with @

Reply:
at the moment errors are coming back in the form of a the generic login dialog, I'm not seeing any errors when using `@` in the display name though :thinking: 

![2022-02-25T12:09:55,986486346+00:00](https://user-images.githubusercontent.com/1848238/155713073-fb553c2c-82b5-4759-9fd7-9cb6c3c18189.png)

Reply:
I mixed between display name and userId, my bad.

Reply:
thanks for the screenshot and explanation! I've checked in with product/design, we'd like to do a pass of all the onboarding errors and their copy

I'll create another subtask in #5200 to revisit the error handling here in case there's other types of errors we'd like to capture within the text field instead of dialog and I'll need to track down all the possible error types (and their copy)

Reply:
Ack